openpype/modules/archiver/rest_api.py

```diff
@@ -0,0 +1,431 @@
+    workdir_data = {
+        'project': {'name': project_name},
+        'hierarchy': "/".join(asset["data"]["parents"]),
+        'asset': asset["name"],
+        'task': {'name': task_name},
+    }
+    return workdir_data
+
+def get_dir_path(
+        project_anatomy: Anatomy, asset_name: str, task_name: str, host_name: str
+    ) -> str:
+    """
+    Get path to directory on the file system
+    """
+    workfile_template_key = get_workfile_template_key(
+        task_type=task_name,
+        host_name=host_name,
+        project_name=project_anatomy.project_name
+    )
+    workdir_data = get_workdir_data(project_anatomy.project_name, asset_name, task_name)
+    dir_template = project_anatomy.templates_obj[workfile_template_key]["folder"]
+    dir_path = dir_template.format_strict(workdir_data)
+    return dir_path
+
+def get_file_template(project_anatomy: Anatomy, host_name: str, task_name: str) -> str:
+    """
+    Get template for file name (as defined in project settings)
+    """
+    workfile_template_key = get_workfile_template_key(
+        task_type=task_name,
+        host_name=host_name,
+        project_name=project_anatomy.project_name
+    )
+    file_template = project_anatomy.templates[workfile_template_key]["file"]
+    return file_template
+
+def get_application_dict(application: Application):
+    """
+    Get dictionary from given application instance
+
+    Returns:
+        Dict[str, Any]: Dictionary holding application info
+    """
+    return {
+        "full_name": application.full_name, 
+        "full_label": application.full_label,
+        "executable_paths": [app_exec.executable_path for app_exec in application.executables],
+        "label": application.label,
+        "name": application.name,
+        }
+
+def check_query_parameters(request: Request, needed_query_parameters):
+    """
+    Args:
+        request (Request): Incoming request
+        needed_query_parameters (Tuple[str]): Parameters
+    Raises:
+        Exception: If given parameters are not found in request query
+    """
+    query_params = dict(request.query)
+    for needed_query_parameter in needed_query_parameters:
+        if needed_query_parameter not in query_params:
+            raise Exception(f"Missing query parameter `{needed_query_parameter}`")
+
+
+
+class ProjectsGetterEndpoint(_RestApiEndpoint):
+    async def get(self) -> Response:
+        project_docs = list(get_projects())
+        projects = []
+        for project_doc in project_docs:
+            projects.append({
+                "id": project_doc["_id"],
+                "name": project_doc["name"],
+            })
+        return Response(
+            status=200,
+            body=self.resource.encode(projects),
+            content_type="application/json"
+        )
+
+
+class ProjectGetterEndpoint(_RestApiEndpoint):
+    async def get(self, project_name: str) -> Response:
+        project_doc = get_project(project_name=project_name)
```

Comment:
You declare method as async but which part is async, you don't use `await get_project...`

Reply:
`async` is necessary as requirement of the aiohttp module that imposes the use of coroutines at server side to answer client calls ([see here](https://docs.aiohttp.org/en/stable/web_quickstart.html#run-a-simple-web-server)). It's unrelated to `get_project`